Tests/VSMTests/StateObservingTests/StateObservingTests.swift

```diff
@@ -0,0 +1,168 @@
+//
+//  StateObservingTests.swift
+//  
+//
+//  Created by Albert Bori on 2/27/23.
+//
+
+import Combine
+@testable import VSM
+import XCTest
+
+/// Tests the `StatContainer`'s implementation of `StateObserving` and acts as a base class for other `StateObserving` types to test their desired outcomes
+class StateObservingTests: XCTestCase {
+    var stateObservingSubject: (any StateObserving<MockState>)!
+    var observedState: (() -> MockState)!
+    var observedStatePublisher: AnyPublisher<MockState, Never>!
+    
+    private var subject: any StateObserving<MockState> { stateObservingSubject }
+    private var state: MockState { observedState() }
+    private var statePublisher: AnyPublisher<MockState, Never> { observedStatePublisher }
+    
+    override func setUp() {
+        let stateContainer = StateContainer<MockState>(state: .foo)
+        stateObservingSubject = stateContainer
+        observedState = { stateContainer.state }
+        observedStatePublisher = stateContainer.$state.eraseToAnyPublisher()
+    }
+    
+    override func tearDown() {
```

Comment:
Should you be using the `withError` versions of these setup and tearDown methods here and elsewhere?